src/entries/popup/pages/home/TokenDetails/TokenDetails.tsx

```diff
@@ -287,34 +342,66 @@ function MoreOptions({ token }: { token: ParsedUserAsset }) {
         <AccentColorProvider
           color={token.colors?.primary || token.colors?.fallback}
         >
-          {!isNative && (
-            <DropdownMenuItem
-              symbolLeft="doc.on.doc.fill"
-              onSelect={() => copyAddress(token.address)}
-            >
-              <Text size="14pt" weight="semibold">
-                {i18n.t('token_details.more_options.copy_address')}
-              </Text>
-              <Text size="11pt" color="labelTertiary" weight="medium">
-                {truncateAddress(token.address)}
-              </Text>
+          {!hidden && (
+            <DropdownMenuItem symbolLeft="pin.fill" onSelect={togglePinToken}>
+              <TextOverflow weight="semibold" size="14pt">
+                {pinned
+                  ? i18n.t('token_details.more_options.unpin_token', {
+                      name: token.symbol,
+                    })
+                  : i18n.t('token_details.more_options.pin_token', {
+                      name: token.symbol,
+                    })}
+              </TextOverflow>
             </DropdownMenuItem>
           )}
           <DropdownMenuItem
-            symbolLeft="safari"
-            external
-            onSelect={() => window.open(getCoingeckoUrl(token), '_blank')}
+            symbolLeft="eye.slash.fill"
+            onSelect={toggleHideToken}
           >
-            CoinGecko
+            <TextOverflow weight="semibold" size="14pt">
+              {hidden
+                ? i18n.t('token_details.more_options.unhide_token', {
+                    name: token.symbol,
+                  })
+                : i18n.t('token_details.more_options.hide_token', {
+                    name: token.symbol,
+                  })}
+            </TextOverflow>
           </DropdownMenuItem>
-          {!isNative && explorer && (
-            <DropdownMenuItem
-              symbolLeft="binoculars.fill"
-              external
-              onSelect={() => window.open(explorer.url, '_blank')}
-            >
-              {i18n.t('token_details.view_on', { explorer: explorer.name })}
-            </DropdownMenuItem>
+          {swappable && (
+            <>
+              {!isNative && (
+                <DropdownMenuItem
+                  symbolLeft="doc.on.doc.fill"
+                  onSelect={() => copyAddress(token.address)}
+                >
+                  <Text size="14pt" weight="semibold">
+                    {i18n.t('token_details.more_options.copy_address')}
+                  </Text>
+                  <Text size="11pt" color="labelTertiary" weight="medium">
+                    {truncateAddress(token.address)}
+                  </Text>
+                </DropdownMenuItem>
+              )}
+              <DropdownMenuSeparator />
+              <DropdownMenuItem
+                symbolLeft="safari"
+                external
+                onSelect={() => window.open(getCoingeckoUrl(token), '_blank')}
```

Comment:
we have `goToNewTab` function somewhere, not sure why 
@estebanmino is there a reason to use `chrome.tabs.create` instead of `window.open`?